lib/core/include/irods/irods_random.hpp

```diff
@@ -1,9 +1,20 @@
 #ifndef IRODS_RANDOM_HPP
 #define IRODS_RANDOM_HPP
 
+#include <string>
+
 namespace irods {
     void getRandomBytes( void * buf, int bytes );
 
+    /// \brief Generate a string of size \p _length consisting of randomly generated alphanumeric characters.
+    ///
+    /// \param[in] _length Desired length of the generated string.
+    ///
+    /// \throws irods::exception If \p _length exceeds max_size or on allocation failure.
+    ///
+    /// \since 4.3.1
+    auto generate_random_alphanumeric_string(std::size_t _length) -> std::string;
```

Comment:
Should we constrain the length to a smaller data type?

size_t seems way too large IMO.

Reply:
Open to suggestions on a type

Reply:
Going with `std::int32_t` for now. Allowing for a signed type lets us play defense against negative values a little better and it is still sufficient to hold a decently large string.

Reply:
Sorry on the late response. Maybe even `std::int16_t`. That gives people quite a bit of room for generating random strings.

Alternatively, you could make it a template and let the caller decide.

I'll leave the final decision up to you.

Reply:
I think `std::int16_t` is fine. Like the RNG backend, this can be turned into an option later if we even continue to use this.

Reply:
Or, this thing that I just discovered...
```c++
auto generate_random_alphanumeric_string(std::signed_integral auto _length) -> std::string;
```
Thoughts?

Reply:
I can't see a reason to allow very large string sizes. As long as it can't be misused and exhaust memory, I'm fine with whatever choice you make.

Using concepts like you have in the sample is always cool to see. I don't how helpful it is here though. Perhaps the template idea is pointless since it would allow virtually any range of signed integers (i.e. someone can pass int64_t).

Reply:
Sure, let's just stick with `std:int16_t` then

Reply:
Done